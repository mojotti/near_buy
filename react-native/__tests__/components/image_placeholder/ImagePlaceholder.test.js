import React from 'react';
import { shallow } from 'enzyme';
importÂ ImagePlaceholder from '../../../src/components/image_placeholder/ImagePlaceholder';

describe('<ImagePlaceholder />', () => {
  test('shows placeholder image when image is not loaded', () => {
    const imagePlaceHolder = shallow(<ImagePlaceholder url={'foo.com'} />);

    expect(imagePlaceHolder).toMatchSnapshot();
  });

  test('does not show placeholder image when image is loaded', () => {
    const imagePlaceHolder = shallow(<ImagePlaceholder url={'foo.com'} />);
    imagePlaceHolder.instance()._onLoad();

    expect(imagePlaceHolder).toMatchSnapshot();
  });
});
