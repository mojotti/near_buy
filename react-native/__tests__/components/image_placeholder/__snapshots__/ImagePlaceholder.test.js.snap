// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagePlaceholder /> does not show placeholder image when image is loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    onLoad={[Function]}
    source={
      Object {
        "uri": "foo.com",
      }
    }
    style={
      Array [
        Object {
          "position": "absolute",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<ImagePlaceholder /> shows placeholder image when image is not loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    onLoad={[Function]}
    source={
      Object {
        "uri": "foo.com",
      }
    }
    style={
      Array [
        Object {
          "position": "absolute",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        undefined,
      ]
    }
  >
    <Icon
      allowFontScaling={false}
      color="gray"
      name="image"
      size={12}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
